Homework 5.cpp

```diff
@@ -0,0 +1,164 @@
+//
+//  Homework 5.cpp
+//  StudyFall16
+//
+//  Created by Farida Eid on 7/8/17.
+//  Copyright © 2017 Farida Eid. All rights reserved.
+//
+
+#include <iostream>
+#include <vector>
+#include <map>
+#include <unordered_map>
+#include <stack>
+#include <set>
+
+using namespace std;
+
+class Graph
+{
+private:
+    map<char,set<char>> adjList;
+    unordered_map<char, bool> visited;
+    void topologicalSortRecursive(char, stack<char>&);
+    
+public:
+    Graph();
+    void addEdge(char, char);
+    void topologicalSort(vector<char>&);
+    
+};
+
+// Graph constructor
+// Parameters: No parameters
+Graph::Graph()
+{
+}
+
+//Function to add edge in the graph
+//Parameters: Vertix 1 and Vertix 2
+void Graph::addEdge(char u, char v)
+{
+    visited[u] = false;
+    visited[v] = false;
+    
+    adjList[u].insert(v);
+}
+
+
+
+// Function to perform topological sort
+// Parameters: The vector of characters to save the answer in
+void Graph::topologicalSort(vector<char>& characters)
```

Comment:
Please use pointers for parameters you will modify inside the function, makes the calling code much more readable.